### PR TITLE
Update clang format

### LINKED ISSUE #43: Optimize clang-format template
If this doesn’t look right, [choose a different type.](https://github.com/JDRaftKeeper/RaftKeeper/issues/new)

### Description

1.  regroup and order includes
2. fix some invalid values
3. standard:  c++11 -> c++20
4. break braces in lambda functions
5. column limit 140 -> 120

### PR DESCRIPTION
**Pull request by JackyWoo**
### Which issues of this PR fixes:

This PR try to fix #43 

### Change log:

1. group includes
2. fix some invalid values

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -1,65 +1,59 @@
 BasedOnStyle: WebKit
 Language: Cpp
 AlignAfterOpenBracket: AlwaysBreak
-BreakBeforeBraces: Custom
+BreakBeforeBraces: Allman
```
Comment (nicelulu) [inline .clang-format:4]: why use Allman

File: .clang-format [hunk 2]
```diff
@@ -1,65 +1,59 @@
...
 TabWidth: 4
 IndentWidth: 4
-Standard: Cpp11
+Standard: c++20
 PointerAlignment: Middle
 MaxEmptyLinesToKeep: 2
 KeepEmptyLinesAtTheStartOfBlocks: false
 AllowShortFunctionsOnASingleLine: InlineOnly
-AlwaysBreakTemplateDeclarations: true
+AlwaysBreakTemplateDeclarations: Yes
```
Comment (nicelulu) [inline .clang-format:30]: why Yes

File: src/Service/ConnectionHandler.cpp [hunk 3]
```diff
@@ -63,18 +63,26 @@ ConnectionHandler::ConnectionHandler(Context & global_context_, StreamSocket & s
     , keeper_dispatcher(global_context.getDispatcher())
     , operation_timeout(
           0,
-          Context::getConfigRef().getUInt("keeper.raft_settings.operation_timeout_ms", Coordination::DEFAULT_OPERATION_TIMEOUT_MS) * 1000)
+          Context::getConfigRef().getUInt(
+              "keeper.raft_settings.operation_timeout_ms", Coordination::DEFAULT_OPERATION_TIMEOUT_MS)
+              * 1000)
```
Comment (nicelulu) [inline src/Service/ConnectionHandler.cpp:68]: I prefer the previous format. This code is too long.

**Comment by nicelulu** [comment]
What's wrong with the previous clang format?

File: .clang-format (hunk 1 above)
Comment (JackyWoo) [reply .clang-format:4]: for `{` break

File: .clang-format (hunk 2 above)
Comment (JackyWoo) [reply .clang-format:30]: true is an invalid value

File: src/Service/ConnectionHandler.cpp (hunk 3 above)
Comment (JackyWoo) [reply src/Service/ConnectionHandler.cpp:68]: I change the column limit from 140 to 120.

**Comment by JackyWoo** [comment]
> What's wrong with the previous clang format?

After format, code style tool may report some errors.

**Comment by nicelulu** [comment]
I think the previous format is fine. Doesn't need to be adjusted.

File: src/Service/ForwardRequest.cpp [hunk 4]
```diff
@@ -206,14 +207,16 @@ ForwardRequestFactory & ForwardRequestFactory::instance()
...
-        auto res = std::make_shared<RequestT>();
-        return res;
-    });
+    factory.registerRequest(
+        num,
+        []()
+        {
+            auto res = std::make_shared<RequestT>();
+            return res;
+        });
```
Comment (nicelulu) [inline src/Service/ForwardRequest.cpp:219]: the previous one was better

File: .clang-format [hunk 5]
```diff
@@ -82,10 +76,10 @@ ObjCBlockIndentWidth: 4
...
 ObjCSpaceBeforeProtocolList: true
 PenaltyBreakBeforeFirstCallParameter: 19
-PenaltyBreakComment: 300
-PenaltyBreakFirstLessLess: 120
-PenaltyBreakString: 1000
-PenaltyExcessCharacter: 1000000
+PenaltyBreakComment: 80
+PenaltyBreakFirstLessLess: 80
+PenaltyBreakString: 80
+PenaltyExcessCharacter: 80
```
Comment (nicelulu) [inline .clang-format:82]: Penalty is not really a limit, it's a penalty and we need to find a balance.

Comment (JackyWoo) [reply .clang-format:82]: ok

File: src/Service/NuRaftFileLogStore.h [hunk 6]
```diff
@@ -53,12 +55,12 @@ class LogEntryQueue
 class NuRaftFileLogStore : public nuraft::log_store
 {
     __nocopy__(NuRaftFileLogStore) public : explicit NuRaftFileLogStore(
-         const std::string & log_dir,
-         bool force_new = false,
-         FsyncMode log_fsync_mode_ = FsyncMode::FSYNC_PARALLEL,
-         UInt64 log_fsync_interval_ = 1000,
-         UInt32 max_log_size_ = LogSegmentStore::MAX_SEGMENT_FILE_SIZE,
-         UInt32 max_segment_count_ = LogSegmentStore::MAX_SEGMENT_COUNT);
+                                                const std::string & log_dir,
```
Comment (nicelulu) [inline src/Service/NuRaftFileLogStore.h:58]: format

File: src/Service/ForwardRequest.h [hunk 7]
```diff
@@ -167,20 +170,17 @@ class ForwardRequestFactory final : private boost::noncopyable
         auto opnum = request_for_session.request->getOpNum();
         switch (opnum)
         {
-            case Coordination::OpNum::SessionID:
-            {
+            case Coordination::OpNum::SessionID: {
```
Comment (nicelulu) [inline src/Service/ForwardRequest.h:173]: format

File: src/Service/ConnectionHandler.cpp [hunk 8]
```diff
@@ -62,10 +62,9 @@ ConnectionHandler::ConnectionHandler(Context & global_context_, StreamSocket & s
     , global_context(global_context_)
     , keeper_dispatcher(global_context.getDispatcher())
     , operation_timeout(
-          0,
           Context::getConfigRef().getUInt("keeper.raft_settings.operation_timeout_ms", Coordination::DEFAULT_OPERATION_TIMEOUT_MS) * 1000)
     , session_timeout(
-          0, Context::getConfigRef().getUInt("keeper.raft_settings.session_timeout_ms", Coordination::DEFAULT_SESSION_TIMEOUT_MS) * 1000)
+          Context::getConfigRef().getUInt("keeper.raft_settings.session_timeout_ms", Coordination::DEFAULT_SESSION_TIMEOUT_MS) * 1000)
```
Comment (nicelulu) [inline src/Service/ConnectionHandler.cpp:67]: 0 is removed after being formatted?

Comment (JackyWoo) [reply src/Service/ConnectionHandler.cpp:67]: removed manually

**Comment by nicelulu** [comment]
I believe that clang-format is merely an auxiliary coding tool rather than the entire solution. When certain coding styles conflict with our established code conventions, we should prioritize our code conventions over complete reliance on it.

**Comment by JackyWoo** [comment]
@nicelulu Auto code format tool is necessary for many developers are accustomed to using it. There will be more developers joining the project in the future. So we must make it ready.

**Comment by nicelulu** [comment]
> @nicelulu Auto code format tool is necessary for many developers are accustomed to using it. There will be more developers joining the project in the future. So we must make it ready.

It is necessary. It's just an aid. When there are one or two areas that conflict with our coding habits, we should focus on our original coding style.

File: .clang-format [hunk 9]
```diff
@@ -21,45 +22,39 @@ ExperimentalAutoDetectBinPacking: true
...
 AlignTrailingComments: false
 
 # Not changed:
 AccessModifierOffset: -4
-AlignConsecutiveAssignments: false
+AlignConsecutiveAssignments: None
 AlignOperands: false
 AllowAllParametersOfDeclarationOnNextLine: true
-AllowShortBlocksOnASingleLine: false
+AllowShortBlocksOnASingleLine: Always
```
Comment (nicelulu) [inline .clang-format:57]: Always -> false

**Pull request merged**